src/main.rs

```diff
@@ -17,15 +20,21 @@ fn min_max(a: i16, b: i16) -> (i16, i16) {
     }
 }
 
-fn build_guides(screen: xcb::Rectangle, pt: xcb::Point, width: u16) -> [xcb::Rectangle; 2] {
+fn build_guides(screen: xproto::Rectangle, pt: xproto::Point, width: u16) -> [xproto::Rectangle; 2] {
     [
```

Comment:
I prefer splitting out rectangle into their own definition first since that would make them one line IIRC.